diff --git a/publication_date.module b/publication_date.module
index ac2a8a2..4229afb 100755
--- a/publication_date.module
+++ b/publication_date.module
@@ -63,32 +63,36 @@ function publication_date_node_delete($node) {
 }
 
 
-
+/**
+ * Save publication date and set it to the node.
+ */
 function _publication_date_set_date($node) {
-      // save publication date
-      if ($node->status == 1) {
-        // already in database ?
-        $date = _publication_date_get_date($node->nid);
-        if (!$date) {
-          // no, we insert it
+  if ($node->status == 1) {
+    // Already in database ?
+    $date = _publication_date_get_date($node->nid);
+    if (!$date) {
+      // No, we insert it.
       $id = db_insert('publication_date')
         ->fields(array(
           'nid' => $node->nid,
           'published_at' => empty($node->published_at) ? REQUEST_TIME : $node->published_at,
         ))
         ->execute();
-        }
-      }
-      else {
-        // already in base ?
-        $date = _publication_date_get_date($node->nid);
-        if ($date) {
-          // yes, so we remove it
+    }
+  }
+  else {
+    // Already in base ?
+    $date = _publication_date_get_date($node->nid);
+    if ($date) {
+      // Yes, so we remove it.
       db_delete('publication_date')
         ->condition('nid', $node->nid)
         ->execute();
-      }
+    }
   }
+
+  // Set the value also to be used later on by other modules invoking the hook.
+  $node->published_at = _publication_date_get_date($node->nid);
 }
 
 /**
